test/integration/targets/apt_virtual_pkg_fix/tasks/main.yml

```diff
@@ -0,0 +1,58 @@
+---
+- name: Create directory for test .deb
+  ansible.builtin.file:
+    path: /tmp/virtual-test/DEBIAN
+    state: directory
+
+- name: Create a test .deb file that depends on a virtual package
+  ansible.builtin.shell: |
+    cat > /tmp/virtual-test/DEBIAN/control << EOF
+    Package: virtual-test-pkg
+    Version: 1.0
+    Architecture: all
+    Description: Test package depending on virtual package
+    Depends: mail-transport-agent
+    EOF
+    dpkg-deb --build /tmp/virtual-test /tmp/virtual-test.deb
+  args:
+    creates: /tmp/virtual-test.deb
+
+- name: Remove any existing MTA providers
+  ansible.builtin.apt:
+    name:
+      - exim4
+      - postfix  
+      - sendmail
+    state: absent
+
+- name: Test the patch - install .deb with virtual package dependency
+  ansible.builtin.apt:
+    deb: /tmp/virtual-test.deb
```

Comment:
Here you can specify the temporary location directly. Something similar to https://github.com/ansible/ansible/blob/0cd36ce6d044fe5d0e200cf52a5dfc05a1e46ce6/test/integration/targets/apt/tasks/apt_deb_depend.yml#L11